src/Modules/OrchardCore.Commerce/Services/LocalInventoryProvider.cs

```diff
@@ -81,7 +81,10 @@ private async Task UpdateInventoryAsync(ProductPart productPart, int difference,
             var relevantInventory = inventoryPart.Inventory.FirstOrDefault(entry =>
                 entry.Key == inventoryIdentifier || entry.Key == inventoryRootIdentifier);
 
-            var newValue = relevantInventory.Value + difference;
+            var newValue = relevantInventory.Value + difference < 0 && inventoryPart.AllowsBackOrder.Value
+                ? 0
+                : relevantInventory.Value + difference;
```

Comment:
Why did you clamp it to zero? I thought we agreed that the inventory should still be decreased. Doesn't this lose the back-order information?

Reply:
Isn't a negative value invalid? In the previous discussion I was thinking of a scenario where, for example, inventory is 5 and a customer orders 3 items (I was unsure if that should decrease inventory or just count as back-ordered items).

So the goal with this change was to make it so if the order amount would decrease the inventory below 0, it should be set to 0 instead. I wasn't aware that this had to keep track of the amount of back-ordered items, it never worked that way, but then I assume it should just keep the negative value?

Reply:
[Back-ordering](https://www.investopedia.com/terms/b/backorder.asp#:~:text=A%20backorder%20is%20an%20order%20for%20a%20good%20or%20service%20that%20cannot%20be%20filled%20at%20the%20current%20time%20due%20to%20a%20lack%20of%20available%20supply.) is when you order more than what's on inventory, not that you what's left in stock. If that setting is allowed, then a negative inventory means that the merchant has to order more from their supplier to resolve the inventory deficits.

So like I said, just change [this line](https://github.com/OrchardCMS/OrchardCore.Commerce/blob/6b89d9d2151b44b3fe9557d3038164941d5d2313/src/Modules/OrchardCore.Commerce/Services/LocalInventoryProvider.cs#L85) to also check `inventoryPart.AllowsBackOrder`.

Reply:
Makes sense, I was operating under the assumption that the back-order information is not meant to be stored this way. I'll amend.